moesif-servlet/src/main/java/com/moesif/servlet/BatchProcessor.java

```diff
@@ -0,0 +1,150 @@
+    }
+
+    /***
+     * Method to add event into a queue for batch-based event transfer. It maintains
+     * time when the last event was added to the event queue.
+     * @param event: EventModel
+     */
+    public void addEvent(EventModel event) {
+        try {
+            // Add the event to the queue
+            this.batchQueue.put(event);
+
+            if (this.debug) {
+                String msg = String.format("Event successfully added to queue. Queue Size = %d", this.batchQueue.size());
+                logger.info(msg);
+            }
+
+        } catch (InterruptedException e) {
+            logger.warning("Add event failed. " + e.toString());
+        } catch (Exception e) {
+            logger.warning("Add event failed. " + e.toString());
+        }
+
+    }
+
+    /***
+     * Method to check if job is running or not.
+     * @return boolean
+     *   Returns true if job is running otherwise false.
+     */
+    public boolean isJobRunning() {
+        return this.jobRunning;
+    }
+
+    /***
+     * Task runner method to send events in batches if at least a batch of events available.
+     * It also keeps track of a flag to show if job is in-progress.
+     * If debug has been enabled then it prints how many batches were sent with total events count.
+     */
+    @Override
+    public void run() {
+        // set jobRunning
+        this.jobRunning = true;
+
+        // Return if queue is empty
+        if (this.batchQueue.isEmpty()) {
+            if (this.debug) {
+                logger.info("No events to send.");
+            }
+
+            // reset jobRunning
+            this.jobRunning = false;
+            return;
+        }
+
+        // callback for async createBatchEvents
+        APICallBack<com.moesif.api.http.response.HttpResponse> callBack = new APICallBack<com.moesif.api.http.response.HttpResponse>() {
+
+            @Override
+            public void onSuccess(HttpContext httpContext, HttpResponse httpResponse) {
+                // ignore
+            }
+
+            public void onFailure(HttpContext context, Throwable error) {
+                // ignore
+            }
+        };
+
+        try {
+            // Get all the Events till now
+            List<EventModel> allEventList = new ArrayList<>();
+            this.batchQueue.drainTo(allEventList);
+
+            // Prepare batches and send them.
+            int batchCount = 0;
+            for(int i=0; i< allEventList.size(); i += this.moesifConfig.eventBatchSize, batchCount +=1) {
+                final int endIndex = Math.min(allEventList.size(), i + this.moesifConfig.eventBatchSize);
+                List<EventModel> curEventList = allEventList.subList(i, endIndex);
+                try {
+                    this.moesifApi.getAPI().createEventsBatchAsync(curEventList, callBack);
+                } catch(JsonProcessingException e) {
+                    logger.warning("Failed to send batch events. " + e.toString());
```

Comment:
Could there be other types of exceptions here? Also if there is a batch of events, and if a single event causes parsing exception, will entire batch fail?